src/script/sign.h

```diff
@@ -128,6 +130,14 @@ struct PartiallySignedTransaction
             s << CFlatData(REF(entry.second));
         }
 
+        // Write the unknown things
+        for(auto& entry : unknown) {
+            WriteCompactSize(s, entry.first.size());
```

Comment:
No need for this - vectors implicitly get their length prepended when serializing (unless you use CFlatData).